### ISSUE TITLE
Support .frm-less recovery

### ISSUE THREAD

**Issue by george-lorch** [facebook/mysql-5.6#757]
When importing this patch **_https://github.com/facebook/mysql-5.6/commit/527f3c4af730fbd6fee13790ff5b1e01fcde7d91 : Disable checking tables with #sql during startup_** we discovered that it seems that MyRocks does not support 'auto-discovery' or the ability to recover a table that exists within MyRocks but does not have a readable .frm file.

**Comment by hermanlee** [comment]
What is the expected recovery path if the .frm file is missing? Looking at some InnoDB recovery mechanisms, I see it records the structure of the table in InnoDB's data dictionary. Do we need a feature to dump the contents of what we believe the table structure to be, re-create the table on a different server and copy the .frm file over?

**Comment by george-lorch** [comment]
Yes, MySQL supports an 'auto-discovery' via handlerton 'discover' calls that allows a storage engine to report back to the server if it has some table that it believes should exist but the server/SQL layer for whatever reason doesn't know about. I believe this problem is one that can arise if a crash occurs at some specific points during some DDL operations or from improper backup/copy routines.

TokuDB took the 'short way' out and keeps a binary copy of the .frm data from CREATE/ALTER within its internal dictionary, but, this hack causes a some problems, particularly on upgrade or cross variant migrations where the .frm content and format may be different or data types have different storage requirements.

I logged this here at @laurynas-biveinis request to keep a record of something that we know doesn't exist and that we might need to look into at some point. In theory though, 8.0 should eliminate the need for this functionality with transactional DDL but I am not certain if it eliminated the 'discover' calls.

**Comment by laurynas-biveinis** [comment]
"Recovery" here, as one of the cases, would also mean a drop instead of actual recovery of the orphaned table in RocksDB - a relatively common use case with temp tables?
